### PR TITLE
prevent streamers with no finished streams from showing

### LINKED ISSUE #158: Streamer in Archive dropdown before first stream was live?
Check if a streamer is shown in the dropdown, even though his first stream was not live yet.
It should NOT be listed there before the first stream has finished.

- Add a test to check that behavior

### PR DESCRIPTION
**Pull request by JonPurvis**
Fixes #158 

Streamers were showing as filterable on the Archive page despite having no finished streams, so when you filtered the archive by them, nothing showed up.

### CODE REVIEW

File: tests/Feature/Http/Livewire/StreamListArchiveTest.php [hunk 1]
```diff
@@ -26,11 +27,16 @@
 
 it('shows streamers as dropdown options', function() {
     // Arrange
-    Channel::factory()
+    $channel_a = Channel::factory()
```
Comment (christophrumpel) [inline tests/Feature/Http/Livewire/StreamListArchiveTest.php:30]: Can you please combine the creation of the channel and the stream like: 

```
Channel::factory()
    ->for(Stream::factory()...
```

This way we have less code and see that the stream belongs to the channel.

**Review by christophrumpel — CHANGES_REQUESTED** [review]
Thanks for the PR. Just added one change request.

File: tests/Feature/Http/Livewire/StreamListArchiveTest.php (hunk 1 above)
Comment (JonPurvis) [reply tests/Feature/Http/Livewire/StreamListArchiveTest.php:30]: Not a problem, I'll sort that this week 🙂

Comment (JonPurvis) [reply tests/Feature/Http/Livewire/StreamListArchiveTest.php]: Sorted that @christophrumpel but let me know if it needs some more tweaks

**Review by christophrumpel — APPROVED** [review]

**Comment by christophrumpel** [comment]
Thanks @JonPurvis 👍

**Pull request merged**